File: news_server/modules/source/infrastructure/getters/tencent.ts
/**
 * 腾讯新闻源获取器
 * 从 newsnow/server/sources/tencent.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'

interface TencentResponse {
  ret: number
  msg: string
  data: {
    tabs: {
      id: string
      articleList: {
        id: string
        title: string
        link_info: {
          url: string
        }
        desc: string
      }[]
    }[]
  }
}

export const tencentHotGetter: SourceGetter = async () => {
  const url = 'https://i.news.qq.com/web_backend/v2/getTagInfo?tagId=aEWqxLtdgmQ%3D'
  
  const response = await fetch(url, {
    headers: {
      Referer: 'https://news.qq.com/',
    },
  })
  
  if (!response.ok) {
    throw new Error(`Failed to fetch Tencent: ${response.statusText}`)
  }

  const res = (await response.json()) as TencentResponse
  
  return res.data.tabs[0].articleList.map((news) =>
    NewsItem.fromEntity({
      id: news.id,
      title: news.title,
      url: news.link_info.url,
      extra: {
        hover: news.desc,
      },
    })
  )
}

File: modules/source/infrastructure/getters/tencent.go
package getters

import "context"

func (r *Registry) tencentHot(ctx context.Context) ([]Item, error) {
	var res struct {
		Data struct {
			Tabs []struct {
				ArticleList []struct {
					ID       string `json:"id"`
					Title    string `json:"title"`
					Desc     string `json:"desc"`
					LinkInfo struct {
						URL string `json:"url"`
					} `json:"link_info"`
				} `json:"articleList"`
			} `json:"tabs"`
		} `json:"data"`
	}
	if err := r.getJSON(ctx, "https://i.news.qq.com/web_backend/v2/getTagInfo?tagId=aEWqxLtdgmQ%3D", map[string]string{
		"Referer": "https://news.qq.com/",
	}, &res); err != nil {
		return nil, err
	}
	if len(res.Data.Tabs) == 0 {
		return nil, nil
	}
	var out []Item
	for _, news := range res.Data.Tabs[0].ArticleList {
		out = append(out, Item{ID: news.ID, Title: news.Title, URL: news.LinkInfo.URL, Extra: map[string]any{"hover": news.Desc}})
	}
	return out, nil
}
